### PR TITLE
Debug: activate balance

### PR DESCRIPTION
**Pull request by violog**

### CODE REVIEW

File: internal/service/handlers/activate_balance.go [hunk 1]
```diff
@@ -95,33 +98,16 @@ func ActivateBalance(w http.ResponseWriter, r *http.Request) {
...
-			}
-		}
-
-		if err = addEventForReferrer(r, *evTypeRef, *balance); err != nil {
-			return fmt.Errorf("add event for referrer: %w", err)
-		}
-
-		return nil
+		balance.ReferredBy = &referral.ID
+		return doVerificationEventUpdates(r, *balance)
```
Comment (Zaptoss) [inline internal/service/handlers/activate_balance.go:107]: We need to add referral specific event for referrer even if referre doesn't scan passport

File: internal/service/handlers/verify_passport.go [hunk 2]
```diff
@@ -292,12 +270,12 @@ func fulfillOrClaimPassportScanEvent(r *http.Request, balance data.Balance) erro
 	if err != nil {
 		return fmt.Errorf("get open passport scan event: %w", err)
 	}
-
 	if event == nil {
 		return errors.New("inconsistent state: balance is not verified, event type is active, but no open event was found")
 	}
 
-	if !evTypePassport.AutoClaim {
+	if !evTypePassport.AutoClaim || balance.ReferredBy == nil {
```
Comment (Zaptoss) [inline internal/service/handlers/verify_passport.go:277]: If you follow this order:
1. Create balance without referral code
2. Verify passport
3. Activate balance

Passport scan event will be fulfilled, not claimed.

Suggestion: filtering event by status data.EventOpen and data.EventFulfilled

Comment (violog) [reply internal/service/handlers/verify_passport.go:277]: good catch

File: internal/service/handlers/activate_balance.go (hunk 1 above)
Comment (violog) [reply internal/service/handlers/activate_balance.go:107]: no, we don't :) (logic resolved internally)

**Review by Zaptoss — APPROVED** [review]

**Pull request merged**